File: files/400-499/404/CH-404.py
import pandas as pd

path = "400-499/404/CH-404 Replacement.xlsx"
input = pd.read_excel(path, usecols="B:D", nrows=7, skiprows=2)
test = pd.read_excel(path, usecols="F:H", nrows=7, skiprows=2).rename(columns=lambda c: c.rstrip(".1") if c.endswith(".1") else c)

result = input.copy()
result["Product ID"] = result["Product ID"].str.replace(
	r"^((?:[^X]*X){2}[^X]*)X",
	r"\1-",
	regex=True,
)

print(result.equals(test))
# fourth entry incorrect in test. 

File: files/400-499/404/CH-404.R
library(tidyverse)
library(readxl)

path <- "400-499/404/CH-404 Replacement.xlsx"
input <- read_excel(path, range = "B3:D9")
test <- read_excel(path, range = "F3:H9")

result <- input |>
  mutate(
    `Product ID` = str_replace(
      `Product ID`,
      "^((?:[^X]*X){2}[^X]*)X",
      "\\1-"
    )
  )

all.equal(result, test)
# Fourth string in testing data incorrect.
